quinn-proto/src/packet.rs

```diff
@@ -820,6 +833,23 @@ mod tests {
         check_pn(PacketNumber::U32(1073741823), &[0xff, 0xff, 0xff, 0xff]);
     }
 
+    #[test]
```

Comment:
Oops, thanks for catching all this. So do the new tests cover enough of the new behavior, or is there more we could/should test here?

Reply:
I added tests for the bugs I encountered and then some; no gaps jump out at me. Note that the packet number encoding is changing again slightly in draft 17 due to the length tag moving into the flags field.

Reply:
Sounds good.